### PR TITLE
Add bootstrap code for OMAP (BeagleBone Black) platform

### PR DESCRIPTION
**Pull request by RostakaGmfun**
/cc @rasendubi @forGGe @yverzun @tkhalymon

### CODE REVIEW

File: core/core.c [hunk 1]
```diff
@@ -0,0 +1,4 @@
+int main()
```
Comment (rasendubi) [inline core/core.c]: `int main(void)`

File: toolchains/gcc-arm-none-eabi.cmake [hunk 2]
```diff
@@ -0,0 +1,9 @@
+include(CMakeForceCompiler)
```
Comment (rasendubi) [inline toolchains/gcc-arm-none-eabi.cmake]: You should not name toolchain file `arm-none-eabi`. The name is way too general—I could use arm-none-eabi-gcc to compile for Cortex-M4, but can't use your toolchain.

File: platform/omap/start.s [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    cmp r0, r1
+    beq .entry
+
+.clean:
+    str r2, [r0]
+    add r0, r0, #1
+    cmp r0, r1
+    blo .clean
+
+.extern core_main
```
Comment (rasendubi) [inline platform/omap/start.s]: `core_main` or `main`?

**Comment by RostakaGmfun** [comment]
Added blinky code. Please test it

File: core/core.c [hunk 4]
```diff
@@ -1,4 +1,26 @@
...
 int main(void)
 {
-    while (1);
+    // enable functional clock and GPIO1 module
+    *((volatile uint32_t *)OMAP_PER_CLOCK_GPIO1) = 1 << 18 | 2;
+    //enable specific pin
+    *((volatile uint32_t *)OMAP_GPIO1 + OMAP_GPIO_EN) |= 1 << 21;
+
+    while (1) {
+        *((volatile uint32_t *)OMAP_GPIO1+OMAP_GPIO_SET) |= 1 << 21;
```
Comment (rasendubi) [inline core/core.c]: Add spaces around `+`, so it's less ambiguous.

**Pull request merged**